google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,260 @@
+                + "FROM customer_client "
+                + "WHERE customer_client.level <= 1";
+
+        // Performs a breadth-first search algorithm to build a mapping of managers to their
+        // child accounts. The map keys are manager IDs, and the values are lists of those
+        // managers' direct child accounts, as represented by CustomerClient objects.
+        while (!unprocessedCustomerIds.isEmpty()) {
+          long customerId = unprocessedCustomerIds.get(0);
+          unprocessedCustomerIds.remove(0);
+          // Issues a search request by specifying page size.
+          SearchPagedResponse response =
+              googleAdsServiceClient.search(
+                  SearchGoogleAdsRequest.newBuilder()
+                      .setQuery(query)
+                      .setCustomerId(Long.toString(customerId))
+                      .setPageSize(PAGE_SIZE)
+                      .build());
+
+          // Iterates over all rows in all pages to get all customer clients under the specified
+          // customer's hierarchy.
+          for (GoogleAdsRow googleAdsRow : response.iterateAll()) {
+            CustomerClient customerClient = googleAdsRow.getCustomerClient();
+            // The customer client making the request will be returned in the
+            // search result with a level of 0.
+            if (customerClient.getLevel().getValue() == 0) {
+              // Store the root customer client, which is the first encountered customer client
+              // that has level of 0.
+              if (rootCustomerClient == null) {
+                rootCustomerClient = customerClient;
+              }
+              // The steps below map parent and children accounts. Continue here so that managers
+              // accounts exclude themselves from the list of their children accounts.
+              continue;
+            }
+
+            // For all level-1 (direct child) accounts that are manager accounts, the above
+            // query will be run against them to create a map of managers to their
+            // child accounts for printing the hierarchy afterwards.
+            List<CustomerClient> customerClientChildren = new ArrayList<>();
+            // Updates the customerClientChildren with the list of children accounts if there
+            // exists an entry in the map for the given manager.
+            if (customerIdsToChildAccounts.containsKey(customerId)) {
+              customerClientChildren = customerIdsToChildAccounts.get(customerId);
+            }
+            // Adds the current child account to the list of the given manager's child accounts,
+            // which will be empty if the manager has not yet been added to the map.
+            customerClientChildren.add(customerClient);
+            // Inserts or updates the map for the given manager with the new list of child accounts.
+            customerIdsToChildAccounts.put(customerId, customerClientChildren);
```

Comment:
Overall, I'm fine with a large number of comments in other places of this file.
However, for these lines, I found it's too bulky and rather make the code hard to read.

Please consider using ListMultiMap. I believe it would make this part much simpler.